### PR TITLE
[WIP] Add Education Edition Features

### LINKED ISSUE #478: Option for Education Mode
# ❇ Feature Request
 Can you add option for enable Education Mode in Bedrock Edition? Like when you enable on configuration file like `nukkit.yml`, some chemical blocks and entities will be enable. It will be useful.

### PR DESCRIPTION
**Pull request by GoodLucky777**
***Add feature of https://github.com/PowerNukkit/PowerNukkit/issues/478.***

**TODO:**
- Blocks
  - [x] Colored Torch RG
    - [ ] Runtime Block States (For PowerNukkit Developers)
  - [x] Colored Torch BP
    - [ ] Runtime Block States (For PowerNukkit Developers)
  - [x] Chemistry Tables
     - [ ] UI
     - [x] Add Block Properties & States
  - [x] Camera
    - [ ] Runtime Block States (For PowerNukkit Developers)
  - [x] Element Blocks
    - [x] Unknown
    - [x] Element 1 to 118
      - [ ] Block Color (I can't find it)
  - [x] Glow Stick (it is for item, not placeable)
  - [x] Hardened Glass
  - [x] Hardened Stained Glass
  - [x] Hardened Glass Pane
  - [x] Hardened Stained Glass Pane
  - [x] Heat Block
    - [ ] Add functionality that melt things like snow without light
  - [x] Underwater TNT
  - [x] Underwater Torch
- Items
  - [x] Balloon
    - [x] Spawn Balloon
      - [x] Correct Coordinate (x + 1.0, y + 1.75, z + 0.5)
    - [x] Spawn Leash Knot
      - [x] Connect with spawned balloon
  - [x] Bleach
  - [x] Camera
    - [x] Spawning Feature
  - [x] Compounds
    - [x] Add Names
  - [x] Glow Stick
    - [ ] Add functionality
  - [x] Ice Bomb
    - [x] Add Cooldown (10 ticks)
  - [x] Medicine
    - [x] Add functionality for each medicine
  - [x] Sparkler
    - [ ] Add Functionality
  - [x] Super Fertilizer
    - [x] Sapling Instant Grow
    - [x] Crops Instant Grow
    - [x] Grow Grasses and Flower when it activate grass block
      - [ ] Bigger Radius and etc.
- Entities
  - [x] Balloon
    - [x] Add NBT data (balloon_attached, balloon_max_height, balloon_should_drop)
      - [x] Remove if balloon_max_height > balloon height
    - [x] Add Lift Force
      - [ ] Fix Speed
    - [x] Add Hit force (Add 0.1 motion)
    - [ ] Limit Motion if connected with a lead
      - [ ] Calculate lift force with mass if it is connected with a balloonable mob
  - [x] Ice Bomb
    - [x] Add functionality (Freeze water 3×3×3)
      - [ ] Change like more vanilla behavior (Check collided blocks with 1.1×1.1×1.1 hitbox)
  - [x] NPC
    - [ ] Add functionality (Looking at Player)
    - [ ] UI
  - [x] Tripod Camera
    - [x] Add functionality (When it activated by the player, it will look at the player and disappear)
    - [ ] Fix attack rule
- Server
  - [x] Add Option for Education Edition in nukkit.yml file
    - [ ] Update translation
  - [x] Add Chemistry Packs info
  - [x] If education-edition is true, Resource Pack Manager will load Chemistry Packs
    - [ ] Update translation (it will log nukkit.resources.chemistry.success)
  - [x] Add creative inventory items for education edition
    - [x] creativeitems_educationedition.json
      - [x] Update (1.16.20)
    - [x] Fix Test NullPointer Error
    - [ ] Fix Chemistry Table
  - [x] Add recipes for education edition
    - [x] recipes_educationedition.json
      - [x] Update (1.16.20)
      - [x] Balloon Recipes
      - [x] Hardened Glass Things Recipes
      - [x] Undye Recipes (Bleach)
      - [x] Underwater Torch Recipe
      - [x] Underwater TNT Recipe
      - [x] Colored Torch RG, BP Recipes
      - [x] Glow Stick Recipes
      - [x] Sparkler Recipes
    - [ ] Gunpowder Recipes
    - [ ] Add recipe for Brewing Stand (Medicine Recipe)
- Additional
  - [x] Add TNT block states
    - [x] Add Explode Bit TNT
  - [x] New Explosion (Underwater Explode)
    - [ ] Check this working or not
    - [ ] Optimize
  - [x] New Spawn Reason
    - [x] BALLOON
    - [x] CAMERA
  - [ ] Add Balloonable
    - [ ] Add Mass for balloonable mobs (to calculate lift force when the mob connected with a balloon)
  - [x] Add Leash Knot Entity

### CODE REVIEW

**Comment by joserobjr** [comment]
WoW

File: src/main/java/cn/nukkit/entity/passive/EntityNpc.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import cn.nukkit.entity.EntityInteractable;
+import cn.nukkit.entity.EntityLiving;
+import cn.nukkit.level.format.FullChunk;
+import cn.nukkit.nbt.tag.CompoundTag;
+
+/**
+ * @author good777LUCKY
+ */
+public class EntityNpc extends EntityLiving implements EntityNPC, EntityInteractable {
```
Comment (joserobjr) [inline src/main/java/cn/nukkit/entity/passive/EntityNpc.java:11]: Source code class name conflict with `EntityNPC` in Windows environment, please change to a different name.

**Review by Polyana-Fontes — CHANGES_REQUESTED** [review]
This PR is awesome! 😁

There is a conflict issue that is preventing me from switching to the branch, the class `EntityNpc`  is having source code name conflict with `EntityNPC` because file names in Windows are case-insensitive. Can you rename it?

File: src/main/java/cn/nukkit/entity/passive/EntityNpc.java (hunk 1 above)
Comment (good777LUCKY) [reply src/main/java/cn/nukkit/entity/passive/EntityNpc.java]: I changed npc class name to `EntityNPCEntity`.

**Comment by good777LUCKY** [comment]
How to fix this Test error?

**Comment by joserobjr** [comment]
@good777LUCKY The PR #632 solved the test error

**Comment by good777LUCKY** [comment]
> @good777LUCKY The PR #632 solved the test error

Thank you!

**Comment by GoodLucky777** [comment]
Closing this PR because PN is no longer maintained.

**Pull request closed without merging**